site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,245 @@
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`10` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "i8": 10,
+    "nullable": true
+  }
+}
+```
+
+Then full expression `a > 10` will be a scalar function expression, and will 
+contain the expressions for `a` and `10` in it's `arguments` field:
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 2,
+    "outputType": {
+      "bool": {
+        "nullability": "NULLABILITY_NULLABLE"
+      }
+    },
+    "arguments": [
+      {
+        "value": {
+          "selection": {
+            "directReference": {
+              "structField": {
+                "field": 1
+              }
+            },
+            "rootReference": {}
+          }
+        }
+      },
+      {
+        "value": {
+          "literal": {
+            "i8": 10,
+            "nullable": true
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+The name of the function is defined by the `functionReference`, which points to
+the index of the extension function to use. This is serialized separately, which
+you will see later in serialized plans.
+
+```json
+{
+  "extensionFunction": {
+    "extensionUriReference": 1,
+    "functionAnchor": 2,
+    "name": "gt"
+  }
+}
+```
+
+## Relations
+
+An logical or physical plan is typically represented as a tree of nodes, such
+as filter, project, scan, or join. Each of these nodes is a **Relation** in 
+Substrait.
+
+For example, to read a named table, we use a `read` relation. This looks like:
+
+```json
+{
+  "read": {
+    "baseSchema": {
+      "names": [
+        "b",
+        "a"
+      ],
+      "struct": {
+        "types": [
+          {
+            "string": {
+              "nullability": "NULLABILITY_NULLABLE"
+            }
+          },
+          {
+            "i64": {
+              "nullability": "NULLABILITY_NULLABLE"
+            }
+          }
+        ],
+        "nullability": "NULLABILITY_REQUIRED"
+      }
+    },
+    "namedTable": {
+      "names": [
+        "my_database",
+        "my_table"
+      ]
+    }
+  }
+}
+```
+
+Read relations provide a schema in `baseSchema` as discussed in the prior section.
+The type of read is a named table, so the `namedTable` field is present with 
+`names` containing the list of name segments (`my_database.my_table`).
+
+Read relations are leaf nodes, so other relations will have them nested within.
+For example, a filter relation of (`a > 10`) will look like:
+
+```json
+{
+  "filter": {
+    "input": {
+      "read": ...
+      }
+    },
+    "condition": ...
+  }
+}
+```
+
+## Plans
+
+First, let's look at reading a table:
+
+```sql
+SELECT * FROM my_table;
+```
+
+A plan has a list of relations, and each top one is a root relation. The root
+relation just has one input relation and then the names of the fields.
```

Comment:
These are facts, and they are true, but I'm not sure what the story you are trying to tell is.  Perhaps something like...

Many plans are represented by a single tree of relations.  The root of the tree is the output of the plan and the leaves in the tree are the sources of data.

Howevr, some plans actually contain multiple trees.  One reason for this is when a plan has a common subtree that occurs multiple times throughout the plan.  That subtree can be pulled out and described on its own and then can be referenced in other trees.  This is why a Substrait plan is actually a list of relations.  Only one of these is a "root relation" and it forms the root of the plan.